onnxruntime/core/graph/graph.cc

```diff
@@ -784,9 +784,17 @@ Graph::Graph(const Model& owning_model,
   // process graph inputs first as we want the type/shape from them to be preferred if a graph input
   // has a matching initializer
   for (auto& graph_input : graph_proto_->input()) {
-    if (utils::HasName(graph_input) && utils::HasType(graph_input)) {
-      name_to_type_map[graph_input.name()] = graph_input.type();
-      GetOrCreateNodeArg(graph_input.name(), &graph_input.type());
+    if (utils::HasName(graph_input)) {
```

Comment:
The IR requires that graph inputs/outputs have names. Should we log the cases they don't instead of silently skipping them ?

Reply:
If it doesn't have a name nothing will match it and it can't be used by any node, so I would say it's harmless and not worth the log message as I don't think it invalidates the model.